### DISCUSSION TITLE
Resurfacing Q: How to set admin event expiry time > 365 days(31536000 seconds)

### DISCUSSION THREAD

**Discussion by bmk9727** [keycloak/keycloak#32828] [Q&A] [answered]
In one of our usecase we want to keep Admin events(as history) more than one year. Looks like realm setting -> Admin event Expiration can be set max to 365 days.

- Is there any way to increase that limit without codebase change?

**Comment by ssilvert** [comment]
@bmk9727 I think you might be seeing [this bug](https://github.com/keycloak/keycloak/issues/32834) in the admin UI.  The value is actually being set to over 365 days, but if you leave the page and come back, you won't see any value at all.

https://github.com/keycloak/keycloak/issues/32834

**Reply by bmk9727** [reply]
@ssilvert Thank you so much for sharing this. Yes we are seeing this issue also. 
It's kind of flaky if i set value below 365 days then in most of the case in configuration that value stay as it's but if i set it more than 365 days then it simply not setting the value more than 365 because when i check admin event it was showing `adminEventsExpiration: 31536000` only.

What we kind of want is to set the admin event expiry time more than 1year.

**Comment by bmk9727** [comment]
Resurfacing this question.
Do we know How to set admin event expiry time > 365 days(31536000 seconds)?
It would be great as my team wanted to keep this **admin event** history for 5years.

**Reply by ssilvert** [reply]
@bmk9727 This should be fixed in the nightly build: https://github.com/keycloak/keycloak/issues/32834

**Reply by bmk9727** [reply]
@ssilvert So that mean we can set any limit for admin event expiry!! Is that correct?
with https://github.com/keycloak/keycloak/issues/32834 i thought it's related to UI issue and not related to setting expirt time more that 1year!!

**Reply by ssilvert** [reply, marked as answer]
Within reason, you can set any limit.  Of course there is the limit on the size of the value itself as it is stored.

**Reply by bmk9727** [reply]
@ssilvert Thank you,
What is the maximum limit for the values in days?
